pkgs/build-support/fetchgithub/default.nix

```
@@ -24,7 +24,7 @@ let
   };
   passthruAttrs = removeAttrs args [ "owner" "repo" "rev" "fetchSubmodules" "forceFetchGit" "private" "githubBase" "varPrefix" ];
   varBase = "NIX${if varPrefix == null then "" else "_${varPrefix}"}_GITHUB_PRIVATE_";
-  useFetchGit = fetchSubmodules || (leaveDotGit == true) || deepClone || forceFetchGit || (sparseCheckout != "");
+  useFetchGit = fetchSubmodules || (leaveDotGit == true) || deepClone || forceFetchGit || (sparseCheckout != (if builtins.isString sparseCheckout then "" else []));
```

It'd be simpler to just do: 

```suggestion
  useFetchGit = fetchSubmodules || (leaveDotGit == true) || deepClone || forceFetchGit || !(sparseCheckout == [ ] || sparseCheckout == "");
```

We could also not even check for `""`, but I suppose someone might be explicitly overriding `sparseCheckout` to `""`, in which case it is useful.